src/metrics/instance_definitions_test.go

```diff
@@ -7,30 +7,54 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func Test_generateInstanceDefinitions90(t *testing.T) {
-	v := semver.MustParse("9.0.0")
-	queryDefinitions := generateInstanceDefinitions(&v)
-
-	assert.Equal(t, 1, len(queryDefinitions))
-}
-
-func Test_generateInstanceDefinitions91(t *testing.T) {
-	v := semver.MustParse("9.1.0")
-	queryDefinitions := generateInstanceDefinitions(&v)
-
-	assert.Equal(t, 2, len(queryDefinitions))
-}
-
-func Test_generateInstanceDefinitions92(t *testing.T) {
-	v := semver.MustParse("9.2.0")
-	queryDefinitions := generateInstanceDefinitions(&v)
-
-	assert.Equal(t, 3, len(queryDefinitions))
-}
-
-func Test_generateInstanceDefinitions10(t *testing.T) {
-	v := semver.MustParse("10.2.0")
-	queryDefinitions := generateInstanceDefinitions(&v)
-
-	assert.Equal(t, 3, len(queryDefinitions))
+func Test_generateInstanceDefinitions(t *testing.T) {
+	tests := []struct {
+		name           string
+		version        string
+		expectedLength int
```

Comment:
I know it is tricky because we use an array of pointers but I wonder if it would be possible to check if the definitions are actually the expected. If we accidentally swap the first and second elements in the `versionDefinitions` array, the current test wouldn't fail.

Reply:
Test now checks if the definitions are actually the expected defs.
Out of order definitions will now fail.